plugins/woocommerce-blocks/assets/js/extensions/payment-methods/cod/index.js

```diff
@@ -44,10 +44,9 @@ const Label = ( props ) => {
  * @return {boolean}  True if COD payment method should be displayed as a payment option.
  */
 const canMakePayment = ( { cartNeedsShipping, selectedShippingMethods } ) => {
-	if ( ! settings.enableForVirtual && ! cartNeedsShipping ) {
-		// Store doesn't allow COD for virtual orders AND
-		// order doesn't contain any shippable products.
-		return false;
+	if ( settings.enableForVirtual && ! cartNeedsShipping ) {
```

Comment:
This bug was hiding COD completely for virtual orders. The shipping logic beneath was still running.